ssh/tailssh/tailssh.go

```diff
@@ -198,6 +212,19 @@ func (srv *server) handleSSH(s ssh.Session) {
 	srv.handleAcceptedSSH(ctx, s, ci, lu)
 }
 
+func (srv *server) handleSessionTermination(ctx context.Context, s ssh.Session, ci *sshConnInfo, cmd *exec.Cmd) {
+	<-ctx.Done()
+	err := ctx.Err()
+	if serr, ok := err.(SSHTerminationError); ok {
+		msg := serr.SSHTerminationMessage()
+		if msg != "" {
+			io.WriteString(s.Stderr(), "\r\n\r\n"+msg+"\r\n\r\n")
+		}
+	}
+	srv.logf("terminating SSH session from %v: %v", ci.src.IP(), err)
+	cmd.Process.Kill()
```

Comment:
It'd be nice to avoid this on Linux if it's already exited nicely on its own. Because on Linux (until `pidfd`: https://lwn.net/Articles/801319/) the Kill here can kill an unrelated process in the future, as I don't think Go yet uses pidfds, so if this pid recycles, it could be a different process.

Instead of passing in a cmd, can you pass in a `killProc func()` and then have the caller pass in a func that's disarmed (perhaps via a sync.Once shared between this goroutine and the post-Wait cleanup).

Then document this method too.

Reply:
I did another way, PTAL